### ISSUE TITLE
GitHub:)

### ISSUE THREAD

**Issue by oryoruk** [greenelab/computational-reagents#9]
GitHub can help with:
- transparency
 - e.g. making scripts, codes public in a public github repo.
- reproducibility
 - e.g. having the commit history can others to use the exact version of a program in reproducing an analysis.

It addresses some of the issues with transparency and reproducibility, but not all so it's not sufficient.

**Comment by gwaybio** [comment]
Agreed! 👍 Paired with [zenodo](https://zenodo.org/), it makes a powerful and citable versioned resource.

Although I have heard talks that Github is becoming too much of a monopoly and that if it ever went under for an extended period of time or was terminated for whatever reason, all benefits are obviously lost.

**Comment by oryoruk** [comment]
wow, zenodo is so cool. thank you Greg.

>  if it ever went under for an extended period of time or was terminated for whatever reason, all benefits are obviously lost.

with this user base, and underlying [git's](https://goo.gl/k6TRGo) open nature, I doubt that.
as long as you have your repo you can host it somewhere else. if they go down, I suspect bitbucket, others will fill in the void.

having said that with other tools on top, GitHub is more than its core functionality now. and that will take a little longer to develop for sure.

**Comment by gwaybio** [comment]
> with this user base, and underlying git's open nature, I doubt that.

Yeah, me too, I _hope_ its here to stay! 

I think a much more likely scenario where github fails alone is when a github repo associated with a manuscript is deleted, updated, moved, or altered in any way that separates it from the paper submission. Sure a user can roll back commits to the submission date, but not in all situations...

**Comment by oryoruk** [comment]
Fair point. How can we make a persistent snapshot of a repo? :thinking: 
I love @greenelab's GitHub enthusiasm! 💯

**Comment by gwaybio** [comment]
We use github releases tied to a zenodo DOI. This may have some similar limitations though. 

There are probably other solutions too!

**Comment by bemert** [comment]
First time using GitHub with this assignment. Seems pretty user friendly, but haven't yet tried using/commiting to more extensive scripts. Agree that it supports transparency, reproducibility, and collaboration. As mentioned by Greg and Onur above, having a permanent copy of one's work is important for research and may not be provided by Github.

**Comment by sklasfeld** [comment]
I have found github useful for both collaborations as well as doing my own personal projects. I like the history feature because I tend to change parts of scripts that I sometimes need to have not changed. Looking through the history, it is easy to fix mistakes using Github. However, I am still trying to learn the difference between branches and forks. Like most of the tools listed, there is a learning curve to getting the commands on the command line correct. For example, the difference between a commit and push is subtle.

**Comment by zz327** [comment]
Github is an awesome tool for colaberation and managing your own script. It could even generate website. However, I feel like for new user, the learning curve is quite steep. You have to keep practice to get use to it.
